examples/poisson.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import shap
+import gc
+from ngboost import NGBRegressor
+from ngboost.distns import Poisson
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
+from scipy.stats import poisson as dist
+
+if __name__ == "__main__":
+
+    SEED = 12345
+    np.random.seed(SEED)
+
+    """
+    This NFL dataset comprises season-long rolling averages of passing stats. It's not an expanding window, so early season games
+    rely on data from the previous year. NFL passing touchdowns closely mimic a Poisson distribution. There is plenty of room for 
+    improvement, this is just an example. 
+
+    For more detailed NFL data, visit https://github.com/mrcaseb/nflfastR
+    """
+
+    url = 'https://raw.githubusercontent.com/btatkinson/sample-data/master/nfl_tds.csv'
+    df = pd.read_csv(url, error_bad_lines=False)
+
+    x = df.pass_touchdown.values
+
+    mean = np.mean(x)
+    variance = np.var(x)
+
+    print("Passing touchdowns mean: {:.4f}".format(mean))
+    print("Passing touchdowns variance: {:.4f}".format(variance))
+
+    k = np.arange(x.max()+1)
+
+    plt.plot(k, dist.pmf(k, mean)*len(x), 'bo', markersize=9, label='expected tds')
+    sns.distplot(x, kde=False, label='actual tds')
+    plt.title("Naive Poisson Dist Using the Mean")
+    plt.legend(loc='upper right')
+    plt.show()
+
+    def preprocess(df):
+        df = df.drop(columns=['season','game_id','posteam','passer_player_name','passer_player_id','defteam'])
+        
+    #     print(df.isnull().sum())
```

Comment:
I'd remove this. I think this also isn't formatted in black so I would apply black to all the files in your PR

Reply:
I was unaware of black formatting, and now I'm a big fan of it. Thank you for the speedy review!

Reply:
No problem! It comes in handy when your pretty erratic (like me) when it comes to formatting to just do ```black .``` in the root. If your a VSCode user you can also have VSCode run black on save which is pretty nice